worldedit-core/src/main/java/com/sk89q/worldedit/regions/Region.java

```diff
@@ -405,18 +404,33 @@ default IChunkSet processSet(IChunk chunk, IChunkGet get, IChunkSet set) {
                 int by = layer << 4;
                 int ty = by + 15;
                 if (!containsEntireCuboid(bx, tx, by, ty, bz, tz)) {
-                    processExtra = true;
-                    char[] arr = set.load(layer);
-                    for (int y = 0, index = 0; y < 16; y++) {
-                        for (int z = 0; z < 16; z++) {
-                            for (int x = 0; x < 16; x++, index++) {
-                                if (arr[index] != 0 && !contains(x, y, z)) {
-                                    arr[index] = 0;
+                    if (!(set instanceof IntSetBlocks)) {
+                        processExtra = true;
+                        char[] arr = set.loadChars(layer);
+                        for (int y = 0, index = 0; y < 16; y++) {
```

Comment:
Same like in the else block?

Reply:
Refactored duplicate code into a private method with next commit.